### DISCUSSION TITLE
NVIDIA AR SDK's ConvertSurreyFaceModel.exe command can't find shape file

### DISCUSSION THREAD

**Discussion by druuzil** [patrikhuber/eos#331] [Q&A] [answered]
tools\ConvertSurreyFaceModel.exe
--shape=<path>/sfm_shape_3448.bin 
--blend_shape=<path>/expression_blendshapes_3448.bin
--topology=<path>/sfm_3448_edge_topology.json 
--contours=<path>/sfm_model_contours.json 
--ibug=<path>/ibug_to_sfm.txt
--out=<output-path>/face_model0.nvf

None of the above commands do anything.  When I reference the sfm_shape_3448.bin file, it always says it's not a valid shape file.

Example:

ConvertSurreyFaceModel.exe --shape= <path>sfm_shape_3448.bin
No shape file was specified.

What am I doing wrong here?  I'm following the instructions as supplied 100% verbatim but none of the commands work.

**Comment by patrikhuber** [comment, marked as answer]
Hi,

eos doesn't contain a tool called `tools\ConvertSurreyFaceModel.exe`. I think you must be using some different software?

> I'm following the instructions as supplied 100% verbatim

Equally, where are you getting those instructions from? The official eos repo does not contain any instructions like that.

**Reply by druuzil** [reply]
Hi Patrik.

Yeah I was trying to compile the EOS model to use with Nvidia Broadcast Augmented Reality.  I did finally get it working, you apparently have to enter all the flags at once.  I found a guide on another site and was successful with it.

Thanks for the quick response.

**Reply by patrikhuber** [reply]
I see! I found that NVIDIA readme now, and yes it very much looks like you have to enter all the parameters at once. You also have to specify a correct relative or absolute path, I don't think `/sfm_shape_3448.bin` would work on Windows, you'd have to remove the `/`, use `.\` (or `./` might work too, not sure), or prepend it with a folder name or full path.

**Reply by druuzil** [reply]
Here's the video I did with it

https://youtu.be/w-8trRjacR0

**Reply by patrikhuber** [reply]
Cool, thanks for linking, nice to see their AR SDK tracking in action!

I saw your comment below your YouTube video:

> If/when I receive a response from Surrey University to my request for the high-res version of the mask, I may go forward with attempting to make my own texture for it (or maybe try to import something from a Skyrim mod or something to use as a face).

The AR SDK's `tools\ConvertSurreyFaceModel.exe` may or may not work with the high-res SfM - unfortunately that convert tool is a black-box executable without code and we can't see what it's doing (might be worth posting an Issue over on their repo though?). If they haven't hard-coded any values, it may likely work though.

You can also texture the low-res model. As you may know you can either use per-vertex colours to colour the model (where indeed the high-res model would be better, the low-res model's number of vertices will be too low), or texture mapping, which people commonly use. You can use the SFM's UV coordinates for that. You will probably have to do quite a bit of coding to add that to their [`FaceTrack`](https://github.com/NVIDIA/BROADCAST-AR-SDK/blob/master/samples/FaceTrack/) example.
